src/merkle.rs

```diff
@@ -0,0 +1,332 @@
+use std::hash::{DefaultHasher, Hash, Hasher};
+
+/* Internal struct explanations
+We have a vector that stores all vectors with the corresponding hashes
+level 0 is all of the hashes of the data, and the subsequent levels
+are the combined hashes of the "branches" in the tree.
+Levels indicates how many total combination we have and count
+how many leaves we have
+*/
+struct MerkleTree {
+    hashes: Vec<Vec<u64>>,
+    levels: usize,
+    count: usize,
+}
```

Comment:
I think this struct should be public, otherwise the only thing we can do with this crate is to run tests. If that's the only purpose and the idea is to make it private for now, please comment this decision.